### ISSUE TITLE
String compare

### ISSUE THREAD

**Issue by marcelhollerbach** [gf-metro/jaxb#25]
Hello, is it known that here

https://github.com/gf-metro/jaxb/blob/de2ac07156926d062017fa4845184fd16f4b88dd/jaxb-ri/runtime/impl/src/main/java/com/sun/xml/bind/v2/runtime/unmarshaller/Loader.java#L121

two string are compared via == and not equals ? And if yes, what is the intend ?

**Comment by phax** [comment]
You may read http://stackoverflow.com/questions/10578984/what-is-string-interning on `String.intern`. It is okay in this case

**Comment by marcelhollerbach** [comment]
mhmm then there seems to be a problem 

Take a look at this code snipped, when i do 
https://github.com/coolcrowd/object-service/blob/master/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/mturk/MturkRestCommand.java

it works, when i take a String and append the version string, the compare fails. When i debugged this I ended up in this If where the intern is different.

**Comment by phax** [comment]
I think the problem is somewhere else as the method is `reportUnexpectedChildElement`.
Are you sure the generated response matches the underlying XSD?

**Comment by marcelhollerbach** [comment]
yes, it matches the scheme url. I am pretty sure the problem is there... I will try to bring a little usecase where it works.

**Comment by marcelhollerbach** [comment]
** where its not working, so you can see the bug**

**Comment by phax** [comment]
I'm not a Jaxb developer so sorry that I dint dive deeper into it....
